lib/common/client/concerns/service_errors.rb

```diff
@@ -14,7 +14,9 @@ def handle_service_error(error)
     def handle_error(error)
       case error.status
       when 401
-        raise Common::Exceptions::Unauthorized, error_details('The upstream server responded 401 Unauthorized')
+        raise Common::Exceptions::BadGateway, error_details(
+          "Received an unauthorized (401) response from the upstream server, check this error's source for event id"
```

Comment:
NITPICK:

Possibly consider making the message more consistent with the message for 403 below:

"The upstream server responded with 401 Unauthorized"

I think checking the errors source for event id etc might go without saying?